src/Containers/PanelListWrapper/useSequentiallyAddedPanels.ts

```diff
@@ -17,24 +17,48 @@ export const useSequentiallyAddedPanels = (
 
     useEffect(() => {
         if (
-            previousPanelsLength.current===undefined ||
+            previousPanelsLength.current === undefined ||
             previousPanelsLength.current < panels.length
         ) {
+            // add panels
             setTimeout(() => {
                 const panelToAdd = panels[panelIndexCounter];
                 if (panelToAdd) {
-                    setInternalPanels((prev) => [...prev, panelToAdd]);
+                    setInternalPanels((prev) => [
+                        ...prev
+                            .filter(
+                                (prevPanel) =>
+                                    prevPanel.name !== panelToAdd.name,
+                            )
+                            .map((prevPanel) => {
+                                const foundPanel = panels.find(
+                                    (panel) => prevPanel.name === panel.name,
+                                );
+                                if (foundPanel) return foundPanel;
+                                return prevPanel;
+                            }),
+                        panelToAdd,
+                    ]);
                     setPanelIndexCounter((prev) => prev + 1);
                 } else {
                     previousPanelsLength.current = panels.length;
                 }
             }, delay);
         } else {
-            setInternalPanels((prev) =>
-                prev.filter((internalPanel) =>
-                    panels.some((panel) => panel.name === internalPanel.name),
-                ),
-            );
+            // remove panels
+            setInternalPanels((prev) => [
+                ...prev
+                    .filter((prevPanel) =>
+                        panels.some((panel) => panel.name === prevPanel.name),
+                    )
+                    .map((prevPanel) => {
+                        const foundPanel = panels.find(
```

Comment:
Why do you need the map? Abs filter?

Reply:
because for one side I have prevPanel, which are the panels already in state (internalPanels). and for the other side I have panels coming from panels prop, which are updated panels in state. so, if panels change, I need to update prevpanels with that change, so I do a map because I return a new element (a panel coming from panels prop)

Reply:
Why not just do [...Existingarry, newElem]

Reply:
because then panels are not updated (when operationSate changes)